Mage.Sets/src/mage/cards/c/ConspicuousSnoop.java

```diff
@@ -37,7 +37,7 @@ public ConspicuousSnoop(UUID ownerId, CardSetInfo setInfo) {
         this.addAbility(new SimpleStaticAbility(new PlayWithTheTopCardRevealedEffect()));
 
         // You may cast Goblin spells from the top of your library.
-        this.addAbility(new SimpleStaticAbility(new PlayTheTopCardEffect(TargetController.YOU, filter, false)));
+        this.addAbility(new SimpleStaticAbility(new PlayFromTopOfLibraryEffect(filter)));
```

Comment:
Dryad Arbor with Arcane Adaptation chosing goblin.
Well, having a clear cast/play distinction is needed for avoiding wrong edge cases. Not even that big of a change.